include/zephyr/acpi/acpi.h

```diff
@@ -117,9 +126,30 @@ int acpi_device_type_get(ACPI_RESOURCE *res);
  *
  * @param signature pointer to the 4-character ACPI signature for the requested table
  * @param inst instance number for the requested table
- * @param acpi_table pointer to the acpi table
+ * @return acpi_table pointer to the acpi table on success else return NULL
+ */
+void *acpi_table_get(char *signature, int inst);
+
+/**
+ * @brief retrieve acpi MAD table for the given type.
+ *
+ * @param type type of requested MAD table
+ * @param tables pointer to the MAD table
+ * @param num_inst number of instance for the requested table
+ * @param max_inst maximum number of MAD table buffer
  * @return return 0 on success or error code
  */
-int acpi_table_get(char *signature, int inst, void **acpi_table);
+int acpi_get_madt(int type, struct acpi_subtable_header **tables, int *num_inst, int max_inst);
```

Comment:
acpi sub-tables are all contiguous and persistent. Why one would need to allocate another place-holder, with the risk of over-allocating (memory loss) and under-allocating (information loss).

Just follow what has been done so far.